source/particles/particle_handler.cc

```diff
@@ -681,6 +724,43 @@ namespace Particles
 
 
 
+  template <int dim, int spacedim>
+  std::map<unsigned int, IndexSet>
+  ParticleHandler<dim, spacedim>::insert_global_particles(
+    const std::vector<Particle<dim, spacedim>> &particles,
+    const std::vector<std::vector<BoundingBox<spacedim>>>
+      &global_bounding_boxes)
+  {
+    // Store the positions in a vector of points, the ids in a vector of ids,
+    // and the properties, if any, in a vector of vector of properties.
+    std::vector<Point<spacedim>>       positions;
+    std::vector<std::vector<double>>   properties;
+    std::vector<types::particle_index> ids;
+    positions.resize(particles.size());
+    ids.resize(particles.size());
+    if (n_properties_per_particle() > 0)
+      properties.resize(particles.size(),
+                        std::vector<double>(n_properties_per_particle()));
+
+    unsigned int i = 0;
+    for (const auto &p : particles)
+      {
+        positions[i] = p.get_location();
```

Comment:
If you only call `.reserve()` instead of `.resize()` above, then you can here do
```suggestion
        positions.emplace_back (p.get_location());
```
and similarly for the next couple of lines, and you don't need the counter `i` at all.

Reply:
But then I cannot reserve the size of the properties all at once, because `reserve` does not allow me to call it with an initializer for the entries.